src/main/java/htsjdk/samtools/reference/FastaSequenceFile.java

```diff
@@ -59,6 +60,16 @@ public FastaSequenceFile(final Path path, final boolean truncateNamesAtWhitespac
         this.in = new FastLineReader(IOUtil.openFileForReading(path));
     }
 
+    /**
+     * Constructs a FastaSequenceFile that reads from the specified input stream (which must not be compressed, i.e.
+     * the caller is responsible for decompressing the stream).
+     */
+    public FastaSequenceFile(final InputStream inputStream, final boolean truncateNamesAtWhitespace) {
+        super((Path) null); // no path
```

Comment:
We need some way of loading the dict file when the path is null, lots of things in gatk want to have a sequence dictionary from the reference.  I think the new constructors should either take _another_  input stream where the user can pass an open stream to the .dict file, or accept a `SAMSequenceDictionary` directly and rely on them loading it separately. (Is there an exposed way to read a dict file anywhere?)

Reply:
I've added a dictionary argument, and added a `loadDictionary(InputStream)` convenience method to `ReferenceSequenceFileFactory`.